### ISSUE TITLE
Offload greedy decoding to the AIU

### ISSUE THREAD

**Issue by rishikakedia** [torch-spyre/torch-spyre#3548]
### What happened?

follow issue in parent issue https://github.com/torch-spyre/torch-spyre/issues/3547

### What did you expect to happen?

Scope

Implement a Top-1/ArgMax reduction on tensors of shape:

[batch, vocab]

where vocab can be 50k–100k+.

Return only the selected token index 
Avoid copying the full logits tensor back to the CPU.

Dependency- Requires extending the current top-k implementation, which today supports only relatively small tensor sizes (≤1024 elements).

### How can we reproduce it?

follow issue in parent issue https://github.com/torch-spyre/torch-spyre/issues/3547

### Any environmental details we need to know?

This was found in current stack with FMS - but intend is to fix in New stack

### Relevant log output

```Shell

```
